### DISCUSSION TITLE
Can this work with non-remarkable PDFs?

### DISCUSSION THREAD

**Discussion by Natedude** [lucasrla/remarks#49] [Q&A]
I have pdfs highlighted in Adobe Acrobat or Preview on macOS. 
Would it be possible to get this to work with those or is a xochitl folder required?

More general questions if anyone can help:
What is the difference between how Remarkable, Acrobat, and Preview store annotations? Do anyone of them save them into the pdf (modifying it)? If so, how?

Any help with this would be greatly appreciated. Thanks.

**Comment by lucasrla** [comment]
In principle, yes, it's possible to tweak parts of this code to extract higlighted text annotated on Adobe Acrobat or Preview. I'm not sure if you have some fluency on Python. If you do, start here: https://github.com/lucasrla/remarks/blob/0f37ca2978ed2ad0b9f1fdc6b8f0627d409b4736/remarks/conversion/text.py#L104

As for your general question, the reMarkable tablet does not modify the PDF, it stores the annotations and higlights on special internal files that we reverse engineer with remarks. Acrobat and Preview are apps that do modify the PDF file when you annotate there. If you're interested in digging deeper into how PDFs are constructed, I suggest you explore the repository and the docs of [PyMuPDF](https://github.com/pymupdf/PyMuPDF/) (on which we heavily depend).
